server/server.ts

```diff
@@ -18,131 +19,108 @@ var user_id = 0;
  * if client hits GET endpoint "localhost:3000/user_id",
  * then let user have a unique id
  */
-app.get('/user_id', (req, res) => {
+app.get('/user_id', (_, res) => {
   user_id++;
   res.end(user_id.toString());
 });
 
 /******************************************************************************/
 
-/**
- * Manages the metadata for each client during a game.
- */
-class UserProps {
-  name!: string;
-  id!: number;
-  socketId!: string;
-  commands!: string;
-
-  constructor(name: string, id: number, socketId: string) {
-    this.name = name;
-    this.id = id;
-    this.socketId = socketId;
-    this.commands = "*";
-  }
-}
-
-let user1: UserProps;
-let user2: UserProps;
+let user1: UserInfo;
+let user2: UserInfo;
 
 /******************************************************************************/
 
 /**
  * Server waits for both players to connect, and sends a "gameplay" event to both
  *  once 2 players have connected
  * Additional players will receive the error event "initiate"
- * Ready to listen for others (imagine that it is a function for one connection)
- * When initializing the game, also initialize state
- *  0 : no commands submitted, 1: P1 only submitted, 2: P2 only submitted, 3: both submitted
  */
-let current_state;  
 io.on('connection', (socket) => {
   console.log('a user connected');
 
   // The "hello" event is how a user initially connects to the server
-  //  msg should contain a userInfo object
-  socket.on('hello', (msg) => { // listening for event `hello`
-    console.log("we received a message: " + msg);
-    msg = JSON.parse(msg);
-
-    // if user is first user, setup
-    if (user1 === undefined || user1.id === msg.id) { 
-      user1 = new UserProps(msg.name, msg.id, socket.id);
-      // Returns 1 to indicate player 1
-      socket.emit("initiate", 1);
-
-    // if user is second user, setup and allow for starting game by telling clients
-    } else if (user2 === undefined || user2.id === msg.id) {
-      user2 = new UserProps(msg.name, msg.id, socket.id);
-      // Returns 2 to indicate player 2
-      socket.emit("initiate", 2);
-
-      socket.emit("gameplay", user1 );
-      socket.to(user1.socketId).emit("gameplay", user2 ); // sends message to specific client based on clientId
-      current_state = 0;
-
-    // for excess users, cya 
+  socket.on('hello', (data: string) => {
```

Comment:
change socket event to 'pairing' rather than 'gameplay' to better reflect actual event occuring